src/convertToObject.js

```diff
@@ -6,7 +6,20 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = sourceString.split(';').map((el) => el.split(':'));
+
+  const result2 = {};
+
+  result.forEach((el) => {
+    if (el.length === 2) {
+      const key = el[0].trim();
+      const value = el[1] ? el[1].trim() : '';
+
+      result2[key] = value;
+    }
+  });
```

Comment:
[CODE PRACTICE] - Instead of using `forEach` to build the `result2` object, consider using `reduce` which is more suitable for transforming an array into an object without the need for an external variable.